experimental/dds/tree2/src/test/feature-libraries/storedToViewSchema.spec.ts

```diff
@@ -142,23 +144,28 @@ describe("storedToViewSchema", () => {
 				assert.equal(nodeSchema.name, key);
 				if (storedNodeSchema.mapFields !== undefined) {
 					// Since its tested separately, assume fields are converted correctly.
-					assert(nodeSchema.mapFields !== undefined);
+					assert(nodeSchema instanceof MapNodeSchema);
 				} else {
-					assert(nodeSchema.mapFields === undefined);
+					assert(!(nodeSchema instanceof MapNodeSchema));
 				}
 
-				assert.equal(nodeSchema.leafValue, storedNodeSchema.leafValue);
-
 				assert.equal(
-					storedNodeSchema.objectNodeFields.size,
-					nodeSchema.objectNodeFields.size,
+					(nodeSchema as Partial<LeafNodeSchema>).leafValue,
+					storedNodeSchema.leafValue,
 				);
 
-				// Since its tested separately, assume fields are converted correctly, and just compare keys to make sure all fields were converted.
-				assert.deepEqual(
-					new Set(storedNodeSchema.objectNodeFields.keys()),
-					new Set(nodeSchema.objectNodeFields.keys()),
-				);
+				if (nodeSchema instanceof ObjectNodeSchema) {
+					assert.equal(
+						storedNodeSchema.objectNodeFields.size,
+						nodeSchema.objectNodeFields.size,
+					);
+
+					// Since its tested separately, assume fields are converted correctly, and just compare keys to make sure all fields were converted.
```

Comment:
Nit
```suggestion
					// Since it's tested separately, assume fields are converted correctly, and just compare keys to make sure all fields were converted.
```